src/Hipace.cpp

```diff
@@ -504,10 +504,12 @@ Hipace::SolveOneSlice (int islice_coarse, const int ibox,
                 const int iby = Comps[WhichSlice::This]["By"];
                 const int nc = Comps[WhichSlice::This]["N"];
                 AMREX_ALWAYS_ASSERT( iby == ibx+1 );
-                m_fields.getSlices(lev, WhichSlice::This).setVal(0., 0, ibx);
-                m_fields.getSlices(lev, WhichSlice::This).setVal(0., iby+1, nc-iby-1);
+                m_fields.getSlices(lev, WhichSlice::This).setVal(
+                    0., 0, ibx, m_fields.m_slices_nguards);
+                m_fields.getSlices(lev, WhichSlice::This).setVal(
+                    0., iby+1, nc-iby-1, m_fields.m_slices_nguards);
             } else {
-                m_fields.getSlices(lev, WhichSlice::This).setVal(0.);
+                m_fields.getSlices(lev, WhichSlice::This).setVal(0., m_fields.m_slices_nguards);
```

Comment:
Here we assume that all MFs in `m_slices` **always** have `m_fields.m_slices_nguards`. If, for some reason, we don't satisfy this (meaning `m_slices[lev][islice].nGrowVect() != m_slices_nguards`), we could get a nasty bug problem. For instance, if the number of guard cells needs to depend on the slice for advanced BC this could be problematic. An easy solution would be to implement a function `Fields::SliceGuardCells()` that would return `m_slices[lev][islice].nGrowVect()` (with proper arguments), or `Fields::setValSlice(lev, slice)`. What do you think?